clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2446,14 +2494,25 @@ void handleResponse(AbstractResponse abstractResponse) {
                     void handleResponse(AbstractResponse abstractResponse) {
                         final OffsetFetchResponse response = (OffsetFetchResponse) abstractResponse;
                         final Map<TopicPartition, OffsetAndMetadata> groupOffsetsListing = new HashMap<>();
-                        for (Map.Entry<TopicPartition, OffsetFetchResponse.PartitionData> entry :
-                                response.responseData().entrySet()) {
-                            final TopicPartition topicPartition = entry.getKey();
-                            final Long offset = entry.getValue().offset;
-                            final String metadata = entry.getValue().metadata;
-                            groupOffsetsListing.put(topicPartition, new OffsetAndMetadata(offset, metadata));
+
+                        if (response.hasError()) {
+                            groupOffsetListingFuture.completeExceptionally(response.error().exception());
+                        } else {
+                            for (Map.Entry<TopicPartition, OffsetFetchResponse.PartitionData> entry :
+                                    response.responseData().entrySet()) {
+                                final TopicPartition topicPartition = entry.getKey();
+                                final Errors error = entry.getValue().error;
+
+                                if (error == Errors.NONE) {
+                                    final Long offset = entry.getValue().offset;
+                                    final String metadata = entry.getValue().metadata;
+                                    groupOffsetsListing.put(topicPartition, new OffsetAndMetadata(offset, metadata));
+                                } else {
+                                    log.warn("Skipping return offset for {} due to error {}.", topicPartition, error);
```

Comment:
It's a little annoying that the API doesn't give us a way to surface these errors somehow. The only way a user can know that the returned offsets are complete is by inspecting the logs. Since the only partition-level error code at the moment seems to be `UNKNOWN_TOPIC_OR_PARTITION`, it may be fine for now, but I'm wondering if we should add something to the API now in case we have more partition-level errors in the future.

Reply:
I could change the `OffsetAndMetadata` to `PartitionData` inside `ListOffsetResponse` directly though I'm not a fan of it since we are not technically exposing `ListOffsetResponse` to end users. LMK.